vllm/model_executor/models/pixtral.py

```diff
@@ -101,50 +103,80 @@ def input_mapper_for_pixtral(ctx: InputContext,
 
     Args:
         ctx: Context of the loaded model.
-        data: data potentially containing image/image embeddings to be mapped
-            to pixel_values in .forward() for a visual QWenLMHeadModel model.
+        data: data potentially containing PIL images to be processed
+            and mapped to `images`.
 
     Returns:
         MultiModalKwargs containing the stacked normalized images tensor or
         image embeddings.
     """
-    # Early exit if we have provided an image to a language only Qwen model
     model_config = ctx.model_config
     tokenizer = cached_get_tokenizer(
         model_config.tokenizer, tokenizer_mode=model_config.tokenizer_mode)
 
     data_list = data if isinstance(data, list) else [data]
 
     images = []
+    image_tokens_list = []
     for image_data in data_list:
         image = ImageChunk(image=image_data)
         encoding = tokenizer.instruct.mm_encoder(image)
         image = torch.from_numpy(encoding.image).to(device="cuda",
                                                     dtype=torch.float16)
         images.append(image)
+        image_tokens_list.append(encoding.tokens)
 
-    return MultiModalKwargs({"images": images})
+    image_tokens = torch.flatten(
+        torch.tensor([
+            token_id for image_tokens in image_tokens_list
+            for token_id in image_tokens
+        ]))
```

Comment:
Is `flatten` necessary here? It looks like you're already doing that in the list comprehension.

Reply:
Good catch - initially I was debugging why it's returning 2D tensor and thought it's coming from here, then I realized it's from the `MultiModalKwargs.batch(mm_inputs)` call but forgot to update back. Let me remove it.